### PR TITLE
Same instance ID in the same transaction bug fixed

### PR DESCRIPTION
**Pull request by AstroChelonian**
1/3 issues resolved. The process exits with an error when two instances with the same ID are added with the same transaction.

remaining tasks: 

Duplication at POST
Duplication at PUT

### CODE REVIEW

**Comment by spl** [comment]
@AstroChelonian Can you please remove the whitespace differences? Also, try to make the diff as small as possible. In the end, I think your diff will just look something like:

```diff
+                         do_or_die(is_set(Ids), error(same_ids_in_one_transaction(Id), _))),
```

You can squash all commits into one.

**Comment by matko** [comment]
This is just one place where double ids have to be checked though. we also need to do this for the case of document replacing, both in the case of a full replace through a POST and for the case of a partial replace through the PUT method.

**Comment by AstroChelonian** [comment]
> This is just one place where double ids have to be checked though. we also need to do this for the case of document replacing, both in the case of a full replace through a POST and for the case of a partial replace through the PUT method.

Thanks @matko I shall address it in ticket 403 :)

File: src/core/api/api_document.pl [hunk 1]
```diff
@@ -177,7 +177,7 @@
 
     with_transaction(Context,
                      forall(
-                         (   json_read_dict_stream(Stream,ID),
+                         (json_read_dict_stream(Stream,ID),
```
Comment (matko) [inline src/core/api/api_document.pl:180]: one more whitespace issue here

**Pull request merged**